content/blog/host-based-volume-encryption-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,249 @@
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  csi.storage.k8s.io/fstype: xfs
+  allowOverrides: hostEncryption
+  hostEncryption: "true"
+  hostEncryptionSecretName: hpe-encrypt
+  hostEncryptionSecretNamespace: hpe-storage
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+Any `PersistentVolumeClaim` created from this `StorageClass` would be encrypted with the below `Secret`.
+
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hpe-encrypt
+  namespace: hpe-storage
+stringData:
+  hostEncryptionPassphrase: "This is a very secret passphrase"
+```
+
+If a user wants to opt-out of encryption for a certain `PersistentVolumeClaim`, the PVC needs to be annotated.
+
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: my-unencrypted-pvc
+  annotations:
+    csi.hpe.com/hostEncryption: "false"
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 64Gi
+```
+
+There are good reasons for allowing certain volumes to be unencrypted as not all data is confidential and encryption does come with both compute overhead and storage inefficiencies (more on this later).
+
+## Namespace controlled host encryption
+
+In scenarios where the namespaced user need to be in control of the key, it's possible for the cluster administrator to delegate the key management. This is incredibly useful when data is being replicated or backed up to an off-site location and data is being reused for dev/test use cases, disaster recovery or running compute heavy analytics to offload production. To allow this behavior, the `StorageClass` needs to be tweaked. Pay attention to `allowOverrides`.
+
+```yaml
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "false"
+  name: hpe-standard-delegated
+provisioner: csi.hpe.com
+parameters:
+  csi.storage.k8s.io/controller-expand-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-expand-secret-namespace: hpe-storage
+  csi.storage.k8s.io/controller-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  csi.storage.k8s.io/fstype: xfs
+  allowOverrides: hostEncryption,hostEncryptionSecretName,hostEncryptionSecretNamespace
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+Now, the user first need to create their own `Secret`.
```

Comment:
Change "need" to "needs". OR, the sentence could read "Now, users first need to create their own 'Secret'." But, since you've used "user" as a singular previously, I recommend the first suggested change.